packages/serialization/json/src/jsonSerializationWriter.ts

```diff
@@ -52,9 +52,10 @@ export class JsonSerializationWriter implements SerializationWriter {
       this.writer.push(JsonSerializationWriter.propertySeparator);
   };
   public writeNumberValue = (key?: string, value?: number): void => {
-    key && value && this.writePropertyName(key);
-    value && this.writer.push(`${value}`);
-    key && value && this.writer.push(JsonSerializationWriter.propertySeparator);
+    const isValuePresent = value !== null && value !== undefined;
+    key && isValuePresent && this.writePropertyName(key);
+    isValuePresent && this.writer.push(`${value}`);
+    key && isValuePresent && this.writer.push(JsonSerializationWriter.propertySeparator);
```

Comment:
Is it easier to handle it like this?
 ```suggestion
    if(isValuePresent){
      key && this.writePropertyName(key);
      this.writer.push(`${value}`);
      key && this.writer.push(JsonSerializationWriter.propertySeparator);
    }
```